### DISCUSSION TITLE
Rich Log xontrib

### DISCUSSION THREAD

**Discussion by AstraLuma** [xonsh/xonsh#4628] [Q&A]
Before I go veering off and writing my own, is there a xontrib to log the shell history to a file?

By which I mean, you know you're about to do something you're going to want to write down, so you can run `richlog start myfile` and it'll note all the commands and output.

Yeah, this is basically an in-process version of `script`.

**Comment by jnoortheen** [comment]
AFAIK there isn't any. As you may know, most of the xontribs are hosted on Github and you can check the xontrib topic or search by xontrib- prefix
